### PR TITLE
Handling target=_blank if link points to a file

### PR DESCRIPTION
**Pull request by AndrewR8**
Issue URL: #753

**Description**:

**Steps to test this PR**:
1. Navigate to http://www.granditaliaperth.co.uk/menus.html and click on the download link for the menu.
2. The download will start and a new blank tab closes immediately

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt [hunk 1]
```diff
@@ -841,6 +835,14 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope, DaxDialogLi
             }
 
             it.setDownloadListener { url, _, contentDisposition, mimeType, _ ->
+                //handling target=_blank if link points to a file
+                if (viewModel.url == null) {     //in an empty tab
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:839]: Could we move this logic to the `ViewModel` and add a test for it?

You'd need to:
- Add a method in the `ViewModel` that handles when `viewmodel.url` is null and calls `viewModel.closeCurrentTab()`
- Create a new `Command` that will be returned from the new method in the `ViewModel`, something like `Command.FileDownload`
- Process that new Command in the `BrowserTabFragment` that calls `destroyWebView()`. 
- And finally add a test for all this logic

Comment (AndrewR8) [reply app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:839]: Thanks for the detailed review @malmstein . I will try to implement this

**Comment by cmonfortep** [comment]
@AndrewR8 one question from my side, is this aligned with the proposed solution in https://github.com/duckduckgo/Android/pull/763? 

My advice is to put on hold any extra change till the assignee reviews this PR. Just in case the approach changes and current review comments don't apply anymore. 👍

**Comment by AndrewR8** [comment]
> @AndrewR8 one question from my side, is this aligned with the proposed solution in #763?

@cmonfortep It seems to me yes.

**Comment by subsymbolic** [comment]
Thanks for submitting this @AndrewR8, I'll review it tomorrow.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt (hunk 1 above)
Comment (subsymbolic) [reply app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:839]: Great feedback @malmstein. As the approach here matches the code around it and potentially fixes a frustrating user bug, I'm happy to accept the PR once we have a solid approach even without this feedback in. The PR will add value, even if it doesn't quite match our ideal internal style.

**Comment by subsymbolic** [comment]
@AndrewR8 thanks for submitting this, it's great to see progress on such a frustrating user bug. There is one issue with this approach however, if multiple tabs are open the dialog appears over the first tab rather than the tab that requested the dialog.

In order to fix this we'll need to coordinate the events happening in the db and make sure the correct tab is selected. Feel free to look into this but it is complex. If we don't hear from you by Friday, we can use the work you have done so far and complete the implementation in-house.

**Comment by AndrewR8** [comment]
@subsymbolic 
Due to this #812 issue there is no way to make sure at this moment  that the correct tab is selected when blank tab closes. May I fix this issue first in this PR? or open a new one?

**Comment by subsymbolic** [comment]
Many thanks for your time on this @AndrewR8 and a big apology that our codebase changed after you had submitted your PR. We have since received and accepted another PR that fixes this in line with our latest changes as it also makes some db updates that will help up with upcoming work. Thanks again for submitting this and for your other contributions in the past.

**Pull request closed without merging**